### DISCUSSION TITLE
Pronouns on my profile

### DISCUSSION THREAD

**Discussion by HunteRoi** [community/community#5768] [Other Feature Feedback, Questions, & Ideas] [answered]
Hey there! As @strugee mentioned in [this issue](https://github.com/isaacs/github/issues/1080) dating from 2017, it would be great to have a dedicated field on the user profile to set our pronouns!

It is something that already exists on other platforms such as [GitLab](https://about.gitlab.com/releases/2021/06/22/gitlab-14-0-released/#set-pronouns-on-gitlab-user-profiles) and I think it would be a great add on GitHub.

What do you think about it? 😄

**Comment by thw0rted** [comment]
I'm surprised not to find more activity here.  I frequently comment on an issue and find myself tripping over pronouns in a reply -- "three posts back, so and so said do it this way, but.... their??? solution only works if...".  Most of the time I mouse over their username and find no details.  Making pronouns a prominent field during signup could help a lot.

**Comment by xomiamoore** [comment]
Just wanted to +1 this as someone with unconventional pronouns. This is becoming more and more standard on other sites with a social component. 

Given that tech in general tends to be biased against women and trans people, it's an act of solidarity to remind folks not to assume pronouns. :)

**Reply by thw0rted** [reply]
Just as a data point, I work in a pretty conservative environment (a military facility), and just today they sent out a building-wide email about correct use of pronouns, apparently in response to a recent Supreme Court case.  Not everybody is going to do it right on the first try, but it's pretty clear which way the train is heading.  It would be great for GH to get on board.

**Comment by stnolting** [comment]
Is there already an (experimental?) option for this that I have not yet discovered? :thinking:

Some days ago there was this GitHub blog post: https://github.blog/changelog/2022-09-23-local-timezones-available-on-profiles/

It seems like the example profile there (see below, [source](https://i0.wp.com/user-images.githubusercontent.com/4021812/191612405-01a07cf4-1280-4e79-9938-27415d0ed4b8.png?w=343&ssl=1)) shows a specific section for pronouns right next to the user name. However, @ashishkeshan's profile does not show this section.

![profile](https://user-images.githubusercontent.com/22944758/192342067-9903d752-688d-4fc8-a8a8-3764081e6d51.png)

**Reply by bspurlin** [reply]
I am against github taking any stand on this politicized, contentious topic.

**Reply by thw0rted** [reply]
What is politicized or contentious here?  This isn't just about people who use unconventional / non-traditional pronouns, it's also very helpful when you're trying to address someone who uses a handle instead of a first name (hi!), or someone from a culture that doesn't have traditionally-gendered names, or someone from a culture whose traditional name genders you are not personally familiar with.

ETA: You have a basically blank profile.  How would I know how to refer to you?  If this thread continued and I had to refer to your post to somebody else, I would probably default to "they" -- "they said it's politicized", etc. -- since I don't know anything about you, and can't guess from your username.

**Reply by stnolting** [reply]
Making pronouns optional (like all those achievement badges) would be a good thing. So everyone could decide on their own if they want to use this feature or not 👍

**Comment by Blacklands** [comment]
I was also surprised to find that this feature does not exist yet.

@stnolting Good find! This section doesn't show for me on the person's profile, either.
Would be nice to get a comment from someone at GitHub on whether this is an internal experimental feature? Could it become a beta feature for us to test, soon?

**Comment by ijsbol** [comment]
Seems that this has been added now ^^

**Reply by thw0rted** [reply]
I don't see it, at least on mobile.

**Reply by Blacklands** [reply]
I don't see it either. :(
Can you post a screenshot, or describe where it can be found?
Or could it be a slow rollout thing, maybe?

**Reply by ppp-x** [reply]
Yesterday I have been able to set my pronouns just fine, however it seems that once you've set them, you can't change them anymore. I wasn't planned to hop genders any time soon but for transgender people or, definitely not uncommon, female programmers who (temporarily) set their pronouns to he/him to avoid sexist responses but later want to change it (back) to she/her, it would be desirable to be able to change pronouns.

**Reply by daira** [reply]
That's a serious limitation; glad I found out about this before setting them. I've changed pronouns multiple times.

**Reply by daira** [reply]
Also, I don't see this at all when I edit my profile. Are they doing an A/B test?

![image](https://user-images.githubusercontent.com/643204/216770799-20735f04-e337-4aec-9c45-c22c4702c529.png)

**Reply by Blacklands** [reply]
> Yesterday I have been able to set my pronouns just fine, however it seems that once you've set them, you can't change them anymore.

That would be _extremely_ disappointing if true.
A significant subset of transgender people go through multiple sets of pronouns before settling on one, while still figuring out their gender and/or how they want to be addressed.
For a subset of non-binary people (specifically genderfluid people, but also others) it's also normal to change pronouns somewhat regularly because they don't _have_ a single fixed gender.
There's also still _new_ pronouns that are being normalized, so people might want to switch to a pronoun that they didn't even know existed yet (or weren't comfortable using yet) when they made their previous choice.
And there's more. I could go on. I hope I don't have to, though.

**Reply by brhx** [reply]
I have to imagine this is why they're A/B testing this feature so aggressively. There's no way this is how it will actually work once it lands, surely.

**Reply by ijsbol** [reply]
I was able to set mine yesterday & got a single screenshot -- though it seems the ability has been removed from my account since.
![firefox_A5Ox2PXKad](https://user-images.githubusercontent.com/57773965/217054857-d563504b-d45e-4d69-a617-94194d3c6996.png)

**Reply by Blacklands** [reply]
Is it a drop-down with a couple of fixed choices, or can you type something custom, too? And could you list the pre-made choices, please?

**Reply by norwd** [reply]
> I was able to set mine yesterday & got a single screenshot -- though it seems the ability has been removed from my account since. ![firefox_A5Ox2PXKad](https://user-images.githubusercontent.com/57773965/217054857-d563504b-d45e-4d69-a617-94194d3c6996.png)

That "Your pronouns will be visible ... including to where local laws restrict using pronouns other than those assigned at birth." part is a major yikes for me 😨. Being from Russia originally this would immediately preclude me from using this feature even though I'm cis. The whole point of the concept would be for inclusivity but essentially flagging lgbtq people for abuse in their home-countries is a concern for me. Maybe there could be a scoping option, like *only show to people I follow* or *only show in countries that are not insane*?

**Reply by norwd** [reply]
> *only show in countries that are not insane*

On second thought this should be the default.

**Reply by ijsbol** [reply]
I agree with that, though still; surely countries that are inclined to use this information in that way would understand how to use a VPN? Personally, I like the idea of being able to select who is able to see them; e.g. Public, Only people who follow me, Only people who I follow & Private.

**Reply by ppp-x** [reply]
> _only show in countries that are not insane_

That option should be labelled exactly that

**Reply by Sqaaakoi** [reply]
This wording is extremely dystopian and tt feels genuinely disrespecful that this disclaimer is so prominent. GitHub ideally should not remind its users that their identity is being made illegal.

**Reply by thw0rted** [reply]
It's a warning, to make sure that residents of oppressive countries don't put themselves in danger without realizing what they've done. I would argue that it's better to err on the side of safety over "disrespect".

**Reply by brhx** [reply]
Also with GitHub being owned by Microsoft I can see them wanting to put more emphasis on this. They're one of the more inclusive and progressive tech companies around, but they're also massive and fully globalised.

I have no doubt that one of the reasons it took so long to land this feature was trying to figure out how to square away the wording on that disclaimer. You probably can't blanket disable the feature for regions where there are legal restrictions on using different pronouns than your AGAB, because it's still useful for cis people.

Perhaps they could add a "only show in countries that are not insane" option, but I can see that adding quite a bit of complexity to the feature. Maybe they could add it one day, but on balance I reckon I'd rather have pronoun support now without it, than pronoun support much later on with that feature included 🤷‍♀️ 

Just my two cents though

**Reply by brhx** [reply]
Huh. The warning has now been removed for me:

<img width="353" alt="image" src="https://user-images.githubusercontent.com/8385528/223488056-9d2a5377-3ac4-4be5-a69f-88c04677dace.png">

**Reply by strugee** [reply]
I can reproduce (in the United States, FWIW). This feels... potentially dangerous to folks in oppressive countries?

**Reply by thw0rted** [reply]
Maybe they removed the warning because they were able to actually disable the feature by region? 🤞🏼

**Comment by brhx** [comment, marked as answer]
It's finally live!! 🎉

https://github.blog/changelog/2023-03-01-add-pronouns-to-your-github-profile/

**Comment by minenwerfer** [comment]
No one cares about your pronouns, kid, this is a serious website, not Reddit.

**Reply by ppp-x** [reply]
No one cares about your opinion on this matter, kid, this is a serious website, not Truth Social
